src/ripple/rpc/handlers/Submit.cpp

```diff
@@ -116,15 +116,8 @@ doSubmit(RPC::JsonContext& context)
         }
     }
 
-    std::string reason;
-    auto tpTrans = std::make_shared<Transaction>(stpTrans, reason, context.app);
-    if (tpTrans->getStatus() != NEW)
-    {
-        jvResult[jss::error] = "invalidTransaction";
-        jvResult[jss::error_exception] = "fails local checks: " + reason;
-
-        return jvResult;
-    }
+    auto tpTrans = std::make_shared<Transaction>(stpTrans);
+    assert(tpTrans);
```

Comment:
If `std::make_shared` fails, it'll throw, so if the execution reaches the `assert`, then `tpTrans` will be set. I think it makes more sense to `assert(tpTrans->getStatus() == NEW);`, or since `NEW` is the default ctor parameter, remove the assert entirely.